rosidl_typesupport_opensplice_cpp/resource/srv_ServiceTypeSupport.cpp.template

```diff
@@ -31,20 +42,271 @@ class DataReader;
+    const void * untyped_ros_response)
+{
+    rosidl_typesupport_opensplice_cpp::Sample<@(spec.pkg_name)::dds_::@(spec.srv_name)Response_> response;
+    const @(spec.pkg_name)::@(spec.srv_name)Response & ros_response = *(reinterpret_cast<const @(spec.pkg_name)::@(spec.srv_name)Response *>(untyped_ros_response));
+    @(spec.pkg_name)::type_support::convert_ros_message_to_dds(ros_response, response.data());
+
+    const rmw_request_id_t & request_header = *(reinterpret_cast<const rmw_request_id_t *>(untyped_ros_request_header));
+
+    rosidl_typesupport_opensplice_cpp::Responder<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> * responder = reinterpret_cast<rosidl_typesupport_opensplice_cpp::Responder<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> *>(untyped_responder);
+
+    responder->send_response(request_header, response);
+}
+
+bool take_response__@(spec.srv_name)(void * untyped_requester, void * untyped_ros_request_header,
+                                     void * untyped_ros_response)
+{
+  @(spec.pkg_name)::@(spec.srv_name)Response & ros_response = *(@(spec.pkg_name)::@(spec.srv_name)Response *)untyped_ros_response;
+  rosidl_typesupport_opensplice_cpp::Requester<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> * requester = reinterpret_cast<rosidl_typesupport_opensplice_cpp::Requester<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> *>(untyped_requester);
+
+  rosidl_typesupport_opensplice_cpp::Sample<@(spec.pkg_name)::dds_::@(spec.srv_name)Response_> response;
+  bool received = requester->take_response(response);
+  if(received)
+  {
+    rmw_request_id_t & req_id = *(reinterpret_cast<rmw_request_id_t *>(untyped_ros_request_header));
+
+    req_id.sequence_number = response.sequence_number_;
+
+    @(spec.pkg_name)::type_support::convert_dds_message_to_ros(response.data(), ros_response);
+    return true;
+  }
+  return false;
+}
+
+void destroy_requester__@(spec.srv_name)(void * untyped_requester)
+{
+  rosidl_typesupport_opensplice_cpp::Requester<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> * requester = reinterpret_cast<rosidl_typesupport_opensplice_cpp::Requester<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> *>(untyped_requester);
+  delete requester;
+}
+
+void destroy_responder__@(spec.srv_name)(void * untyped_responder)
+{
+  rosidl_typesupport_opensplice_cpp::Responder<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> * responder = reinterpret_cast<rosidl_typesupport_opensplice_cpp::Responder<@(spec.pkg_name)::dds_::@(spec.srv_name)Request_, @(spec.pkg_name)::dds_::@(spec.srv_name)Response_> *>(untyped_responder);
```

Comment:
These lines are super long, it's really painful to read them in this diff. Can you try to shorten them?